packages/@aws-cdk/aws-events/lib/event-bus.ts

```diff
@@ -0,0 +1,223 @@
+  }
+
+  /**
+   * Import an existing event bus resource
+   * @param scope Parent construct
+   * @param id Construct ID
+   * @param attrs Imported event bus properties
+   */
+  public static fromEventBusAttributes(scope: Construct, id: string, attrs: EventBusAttributes): IEventBus {
+    class Import extends Resource implements IEventBus {
+      public readonly eventBusArn = attrs.eventBusArn;
+      public readonly eventBusName = attrs.eventBusName;
+      public readonly eventBusPolicy = attrs.eventBusPolicy;
+      public readonly eventSourceName = attrs.eventSourceName;
+    }
+
+    return new Import(scope, id);
+  }
+
+  private static eventBusProps(defaultEventBusName: string, props?: EventBusProps) {
+    if (props) {
+      const { eventBusName, eventSourceName } = props;
+      const eventBusNameRegex = /^[\/\.\-_A-Za-z0-9]{1,256}$/;
+
+      if (eventBusName !== undefined && eventSourceName !== undefined) {
+        throw new Error(
+          `'eventBusName' and 'eventSourceName' cannot both be provided`
+        );
+      } else if (eventBusName !== undefined) {
+        if (eventBusName === 'default') {
+          throw new Error(
+            `'eventBusName' must not be 'default'`
+          );
+        } else if (eventBusName.indexOf('/') > -1) {
+          throw new Error(
+            `'eventBusName' must not contain '/'`
+          );
+        } else if (!eventBusNameRegex.test(eventBusName)) {
+          throw new Error(
+            `'eventBusName' must satisfy: ${eventBusNameRegex}`
+          );
+        }
+        return { eventBusName };
+      } else if (eventSourceName !== undefined) {
+        // Ex: aws.partner/PartnerName/acct1/repo1
+        const eventSourceNameRegex = /^aws\.partner(\/[\.\-_A-Za-z0-9]+){2,}$/;
+        if (!eventSourceNameRegex.test(eventSourceName)) {
+          throw new Error(
+            `'eventSourceName' must satisfy: ${eventSourceNameRegex}`
+          );
+        } else if (!eventBusNameRegex.test(eventSourceName)) {
+          throw new Error(
+            `'eventSourceName' must satisfy: ${eventBusNameRegex}`
+          );
+        }
+        return { eventBusName: eventSourceName, eventSourceName };
+      }
+    }
+    return { eventBusName: defaultEventBusName };
+  }
+
+  /**
+   * The physical ID of this event bus resource
+   */
+  public readonly eventBusName: string;
+
+  /**
+   * The ARN of the event bus, such as:
+   * arn:aws:events:us-east-2:123456789012:event-bus/aws.partner/PartnerName/acct1/repo1.
+   */
+  public readonly eventBusArn: string;
+
+  /**
+   * The policy for the event bus in JSON form.
+   */
+  public readonly eventBusPolicy: string;
+
+  /**
+   * The name of the partner event source
+   */
+  public readonly eventSourceName?: string;
+
+  constructor(scope: Construct, id: string, props?: EventBusProps) {
+    const { eventBusName, eventSourceName } = EventBus.eventBusProps(
+      Lazy.stringValue({ produce: () => this.node.uniqueId }),
```

Comment:
Is there a way to make this generate a unique id across stacks? I could prefix with `Aws.STACK_NAME` but I didn't see that being done in code anywhere.

Reply:
Yes, it will be unique.